Marina/add split datasource util

marinacpkt opened the pull request:
Add utility scripts to:

* export/import dashboards, datasources and preferences
* split datasources
* convert dashboards from Grafana 9 to 7
* rename field names in dashboards
* remove default template variable values for query types

marinacpkt committed:
Add dashboard transfer scripts

- Add utility scripts to explort dashboards, datasources and preferences from a cclear into local files.
- Add utility scripts to import dashboards, datasources and preferences to a cclear from local files.

marinacpkt committed:
Add a util to convert dashboards using multiple datasource.

- Add script take in a folder of dashboards
- Convert each dashboard from using one datasource to using multiple
- The appropriate datasource to used is based on the name of the measurements in the queries.

marinacpkt committed:
Add uitility scripts

dcooper-cpacket commented on utilities/transfer/dashboard_export.sh:
```diff
@@ -0,0 +1,169 @@
+
+show_help() {
+cat << EOF
+Usage: $0 [-u USER] [-p PASSWORD] [-f FROM_FOLDER] [-t TARGET_HOST]
```
Trivial style comment.  Consider moving all of the required before the optional (put -t before -f):
Usage: $0 [-u USER] [-p PASSWORD] [-t TARGET_HOST] [-f FROM_FOLDER]

marinacpkt replied:
Done

dcooper-cpacket commented on utilities/transfer/dashboard_export_updated.sh:
I didn't do a diff with "datasource_export.sh", so I'm probably missing something obvious.

- This seems to export datasources.  The script/file name seems to imply different functionality.
- How does this file differ from datasource_export.sh?

marinacpkt replied:
Fixed by refactoring to have one single script now.

dcooper-cpacket commented on utilities/transfer/preference_import.sh:
Both org and user preferences are exported.  However, only user preferences are imported.  Do we want to provide the option (or a separate script) to import the org preferences?

marinacpkt replied:
Fixed by importing all preferences in the provided folder.

marinacpkt committed:
Update utility scripts

- Update export/import scripts to work with legacy and basic auth in one file for eaiser maintenance
- Update export scripts to inlcude cClear version and Grafana version as exported names.
- Add document and exception handling to utility scripts.
- Removed unused scripts

marinacpkt committed:
Update to remove cookie file after all curl commands are done.

marinacpkt committed:
Remove the unzipped folder after import, otherwise it would reimport on the next interation.
